File: C_Programming/Practise_Programs/PP_147.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_147
//
//  Description       : This program uses a typedef to create an
//                      alias for unsigned int and checks whether
//                      the 3rd bit of a number is ON or OFF.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 18/08/2026
//
//  Time Complexity   : O(1)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

#include<stdio.h> 

typedef unsigned int UINT;

int main()
{
    UINT iNo = 0;
    UINT iMask = 4;
    UINT iAns = 0;

    printf("Enter Number : \n");
    scanf("%d", &iNo);

    iAns = iNo & iMask;

    if(iAns == iMask)
    {
        printf("3rd bit is ON\n");
    }
    else
    {
        printf("3rd bit is OFF\n");
    }

    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      12
//
//  Sample Output :
//
//      Enter Number :
//      3rd bit is ON
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/Practise_Programs/PP_147.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_147
//
//  Description       : This program prints a centered
//                      increasing star pattern using nested
//                      for loops. The number of spaces decreases
//                      while the number of stars increases in
//                      each successive row.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 02/09/2026
//
//  Time Complexity   : O(n^2)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

class PP_147
{
    public static void main(String A[])
    {
        int i, j, k = 0;

        for(i = 1; i <= 5; i++)
        {
            for(j = 5 - i; j >= 1; j--)
            {
                System.out.print(" ");
            }
            for(k = 1; k <= i; k++)
            {
                System.out.print(" *");
            }
            System.out.println();
        }
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      No input required.
//
//  Sample Output :
//
//           *
//          * *
//         * * *
//        * * * *
//       * * * * *
//
//////////////////////////////////////////////////////////////////